client-spark/spark3/src/main/java/org/apache/spark/shuffle/writer/RssShuffleWriter.java

```diff
@@ -442,6 +442,8 @@ public Option<MapStatus> stop(boolean success) {
         return Option.empty();
       }
     } finally {
+      // cancel all async thread task related
+      sendingSet.stream().forEach(eventTask -> eventTask.cancel(true));
```

Comment:
If the task held by this future is still in the queue of dataPusher, the cancel operation will not be valid?

Reply:
And if the task is running of pushing data to shuffle server, the interrupt signal caused by the operation of `future.cancel` will be accepted and then exit quickly, right?

Reply:
> If the task held by this future is still in the queue of dataPusher, the cancel operation will not be valid?

It will mark the result of the runnable, when the implementation `AsyncSupply` of runnable is executed,  since the result is set, directtly skip the execution of   the actual  `Supplier` function

Reply:
> And if the task is running of pushing data to shuffle server, the interrupt signal caused by the operation of `future.cancel` will be accepted and then exit quickly, right?

In fact, the signal will not be passed to the executing thread, i'll change the `CompletableFuture` to ` FutureTask`